src/lib/adapters/WebDav.js

```diff
@@ -0,0 +1,368 @@
+/* @jsx el */
+
+import InputInitializeHook from '../InputInitializeHook'
+import Bookmark from '../Bookmark'
+import humanizeDuration from 'humanize-duration'
+
+const {h} = require('virtual-dom')
+
+function el (el, props, ...children) {
+    return h(el, props, children)
+};
+
+const url = require('url')
+
+function getBookmarkKey (bm) {
+    let key = "";
+
+    key = bm.path;
+    key = "," + bm.title;
+    key = "," + bm.url;
+
+    return key;
+}
+
+export default class WebDavAdapter {
+    constructor (server) {
+        console.log('Webdav constructor');
+        console.log(server);
+
+        this.server = server;
+        this.db = new Map();
+
+        console.log ("THIS");
+        console.log (this);
+    }
+
+    setData (data) {
+        this.server = data;
+    }
+
+    getData () {
+        return JSON.parse(JSON.stringify(this.server));
+    }
+
+    getLabel () {
+        let data = this.getData();
+        return data.username + '@' + data.url;
+    }
+
+    getBookmarksAsJSON () {
+        let bookmarksList = [];
+        let values = Array.from(this.db.values());
+
+        for (var i = 0; i < values.length; ++i)
+        {
+            let value = values [i];
+            bookmarksList.push (
+                {
+                    id: value.id,
+                    path: value.path,
+                    url: value.url,
+                    title: value.title,
+                });
+        }
+
+        return JSON.stringify (bookmarksList, null, 4);
+    }
+
+    async syncComplete () {
+        console.log ("WebDav: Copying JSON file to server");
+        this.bookmarksAsJSON = this.getBookmarksAsJSON ();
+
+        console.log ("path :" + this.server.bookmark_file + ":");
+        console.log ("BODY");
+        console.log (this.bookmarksAsJSON);
+        console.log (this.server);
+
+        let fullUrl = this.server.bookmark_file;
+        fullUrl = this.server.url + fullUrl;
+        console.log ("fullURL :" + fullUrl + ":");
+
+        try {
+            await fetch (fullUrl, {
+                    method: 'PUT',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'Authorization': 'Basic ' + btoa(this.server.username + ':' + this.server.password)
+                    },
+                    body: this.bookmarksAsJSON
+                });
+        } catch (e) {
+            console.log ("Error Caught");
+            console.log (e);
+            throw new Error('Network error: Check your network connection and your account details');
+        }
+    }
+
+    async pullFromServer () {
+        let fullUrl = this.server.bookmark_file;
+        fullUrl = this.server.url + fullUrl;
+        console.log ("fullURL :" + fullUrl + ":");
+
+        let myBookmarks = this.getBookmarksAsJSON ();
```

Comment:
seems to be unused :)

Reply:
I removed it, it will be in (or rather not in :) a later commit.